### ISSUE TITLE
[Flag] -v

### ISSUE THREAD

**Issue by davidjpeacock** [davidjpeacock/kurly#18]
```
       -v, --verbose
              Be more verbose/talkative during the operation. Useful for debugging and seeing what's going on "under the hood". A line  starting  with  '>'  means  "header
              data"  sent  by curl, '<' means "header data" received by curl that is hidden in normal cases, and a line starting with '*' means additional info provided by
              curl.

              Note that if you only want HTTP headers in the output, -i, --include might be the option you're looking for.

              If you think this option still doesn't give you enough details, consider using --trace or --trace-ascii instead.

              This option overrides previous uses of --trace-ascii or --trace.

              Use -s, --silent to make curl quiet.
```

curl reports more data such as TCP level information like server IP etc.  Leaving this issue open to make curly more informative.

**Comment by aki237** [comment]
For that we just need to add a simple `httptrace.ClientTrace` with GotConn to the req with context.
In [line 183 of main.go](https://github.com/davidjpeacock/kurly/blob/master/main.go#L183), we are printing the all the headers even before the actual TCP connection is made.

Remove 183 through 186, and we can add this :
```go
tracer := &httptrace.ClientTrace{
	GotConn: func(connInfo httptrace.GotConnInfo) {
		fmt.Printf("*\tTrying %s...\n", strings.Split(connInfo.Conn.RemoteAddr().String(), ":")[0])
		fmt.Fprintln(Outgoing, req.Method, req.URL.Path, req.Proto)
		for k, v := range req.Header {
			fmt.Fprintln(Outgoing, k, v)
		}
                // HERE
	},
}
req = req.WithContext(httptrace.WithClientTrace(req.Context(), tracer))
resp, err := client.Do(req)
```

ie., moving the header printing after the connection is made.

Say if we have to print stuff about ssl versions and other things do the interface conversion of `connInfo.Conn` (which is a `net.Conn`) to `tls.Conn` and check if it is ok.

```go
// HERE
tlsconn, ok := connInfo.Conn.(*tls.Conn)
if ok {
    // Do all the HTTPS stuff here.
}
```

**Comment by aki237** [comment]
actually [`httptrace.ClientTrace`](https://golang.org/pkg/net/http/httptrace/#ClientTrace) provides many other useful callbacks.

**Comment by davidjpeacock** [comment]
Are you interested in implementing this @aki237?

Thank you for the information either way! 👍

**Comment by aki237** [comment]
Sure. But can you give me the list of all the details to be displayed

eg.,

```
    Trying 213.23.245.134...    // IP
    TLS data like versions and certifying authority etc.,
```

**Comment by davidjpeacock** [comment]
I'll work on that. :-)

**Comment by aki237** [comment]
any details?

**Comment by davidjpeacock** [comment]
Sorry @aki237 - I've been crazy busy - this might have to wait a while. :-)

**Comment by aki237** [comment]
Sure

**Comment by davidjpeacock** [comment]
@aki237 My apologies.

I'm not going to get to this any time soon; I suggest we put this on the back burner, if that's cool?

**Comment by aki237** [comment]
@davidjpeacock Sure...

**Comment by aki237** [comment]
@davidjpeacock  currently we are printing the headers to the `Outgoing` writer, before even making the connection. Can we move the lines through [212](https://github.com/davidjpeacock/kurly/blob/master/main.go#L212) and [215](https://github.com/davidjpeacock/kurly/blob/master/main.go#L215) to the `WroteHeaders` function of the [ClientTrace struct](https://golang.org/pkg/net/http/httptrace/#ClientTrace)?

**Comment by aki237** [comment]
@davidjpeacock I have added some verbosity using the httptrace package functionality. PR #67

**Comment by davidjpeacock** [comment]
Thanks @aki237; I'll check it this weekend. 👍

**Comment by davidjpeacock** [comment]
Closed thanks to #67 

:+1: @aki237
